test/e2e/util/util.go

```diff
@@ -35,7 +36,7 @@ import (
 // This function calls t.Cleanup() with a function which tears down the cluster
 // and *blocks until teardown completes*, so no explicit cleanup from the caller
 // is required.
-func CreateCluster(t *testing.T, ctx context.Context, client crclient.Client, opts cmdcluster.Options, artifactDir string) *hyperv1.HostedCluster {
+func CreateCluster(t *testing.T, ctx context.Context, client crclient.Client, opts *core.CreateOptions, artifactDir string) *hyperv1.HostedCluster {
```

Comment:
This was intentionally using a value type before

Reply:
Can change it that way, only have to change some callback flows. :+1:

In general I trust a `DeepCopy()` more than copy by value type.

Reply:
Agree. I can see now the `CreateOptions` has at least one reference type (a string array), so maybe having `DefaultClusterOptions()` return a `DeepCopy()` would solve it more reliably going forward? What do you think?

Reply:
> Agree. I can see now the `CreateOptions` has at least one reference type (a string array), so maybe having `DefaultClusterOptions()` return a `DeepCopy()` would solve it more reliably going forward? What do you think?

Sounds good. Done.